src/main/java/org/hydev/mcpm/client/models/PluginTrackerModel.java

```diff
@@ -0,0 +1,60 @@
+package org.hydev.mcpm.client.models;
+
+/**
+ * 
+ * Representation of a plugin for use in the local plugin tracker.
+ */
+
+public class PluginTrackerModel {
+    String name;
+    Boolean isManual;
+    String versionId;
+    String pluginId;
+
+    /**
+     * Constructor for PluginTrackerModel.
+     *
+     * @param name      The name of the plugin.
+     * @param isManual  Whether the plugin is manually installed.
+     * @param versionId The version ID of the plugin.
+     * @param pluginId  The plugin ID of the plugin.
+     */
+    public PluginTrackerModel(String name, Boolean isManual, String versionId, String pluginId) {
+        this.name = name;
+        this.isManual = isManual;
+        this.versionId = versionId;
+        this.pluginId = pluginId;
+    }
```

Comment:
It's better to use a record than a class here so we don't need to create getters and setters